node/router/messages/src/helpers/codec.rs

```diff
@@ -75,6 +75,8 @@ impl<N: Network> Decoder for MessageCodec<N> {
             None => return Ok(None),
         };
 
+        UnconfirmedTransaction::<N>::check_size(&bytes)?;
```

Comment:
I'm confused. Why is the UnconfirmedTransaction logic being used by MessageCodec for all message types?

Reply:
Oh, I thought this is what you meant with https://github.com/AleoHQ/snarkOS/pull/3229#discussion_r1583886044? 
This check needs to be done from the codec unless we choose to not use `FromBytes` and `ToBytes` in the codec. This is because we can't know the message type before decoding the message ID and currently that is coupled with the rest of message deserialisation.